release_collection.py

```diff
@@ -80,6 +80,33 @@ def check_versions_updated(cur_ref, new_ref, versions_updated):
             versions_updated[3] = True
 
 
+def comp_versions(cur_ref, new_ref):
+    """Compare versions.
+
+    Return values:
+    1  - if cur_ref > new_ref
+    0  - if cur_ref == new_ref
+    -1 - if cur_ref < new_ref"""
+
+    try:
+        _mobj = re.match(r'v*(\d*.\d*.\d*)', cur_ref)
```

Comment:
Does this mean the current function ``` has an error in that it doesn't handle versions from sshd like `v0.15.2`?
Note that the `StrictVersion` class has `_cmp` - so you can do
```
...
        if _mobj and _mobj.group(1):
            new_ref = _mobj.group(1)
        return StrictVersion(cur_ref)._cmp(StrictVersion(new_ref))
```
The reason why `check_versions_updated` iterates on the `version` array is because it has to know which version changed - the major, minor, or patch version, in order to do the semantic versioning for the collection. If you are just comparing two versions you can just use `_cmp`

Reply:
Thank you, @richm! I'm going to fix my code. BTW, when I try to play with StrictVersion, then I got this warning.
```
$ python --version
Python 3.10.5
$ python
...
>>> from distutils.version import StrictVersion
<stdin>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
```
It seems the new package `setuptools` does the same thing this way. Is it ok to replace `distutils` with `setuptools`?
```
from pkg_resources import parse_version
>>> parse_version("v1.2.3") > parse_version("1.2.2")
True
>>> parse_version("1.2.3") < parse_version("v1.2.2")
False
>>> parse_version("v1.2.3") == parse_version("1.2.2")
False
```

Reply:
> It seems the new package setuptools does the same thing this way. Is it ok to replace distutils with setuptools?

tl;dr - let's do this in a separate PR - note that there is some other code in certificate and firewall that we will have to change too - let's just do it all at once - because we will have to write some custom code for this . . . :-( it seems `packaging` is not built-in to python 3.6
```
>python3.6
Python 3.6.15 (default, Jun 10 2022, 00:00:00) 
[GCC 11.3.1 20220421 (Red Hat 11.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from packaging.version import Version, parse
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'packaging'
```
so I think we will have to create some sort of wrapper class or functions of our own for versioning e.g.
```python
try:
    import packaging.version

    def lsr_parse_version(v_str):
        return packaging.version.parse_version(v_str)

except ImportError:
    import distutils.version

    def lsr_parse_version(v_str):
        # strip leading "v" if any
        return distutils.version.StrictVersion(v_stripped)

def lsr_compare_versions(v1_str, v2_str):
    v1 = lsr_parse_version(v1_str)
    v2 = lsr_parse_version(v2_str)
    # no compare method :-(
    if v1 < v2: return -1
    elif v1 > v2: return 1
    else: return 0
```
Then `check_versions_updated` will be tricky because the array is `release` instead of `version` . . .

Reply:
Ok. I will leave the version handling in a separate PR (including the sshd's special case starting with `v`. Hope it's ok...). Thanks!